src/mod_zone_difficulty_scripts.cpp

```diff
@@ -93,23 +141,399 @@ void ZoneDifficulty::LoadMapDifficultySettings()
+    {
+        do
+        {
+            uint32 creatureentry = (*result)[0].Get<uint32>();
+            float hpmodifier = (*result)[1].Get<float>();
+            bool enabled = (*result)[2].Get<bool>();
+
+            if (enabled)
+            {
+                sZoneDifficulty->CreatureOverrides[creatureentry] = hpmodifier;
+                //LOG_INFO("sql", "New creature with entry: {} has exception for hp: {}", creatureentry, hpmodifier);
+            }
+        } while (result->NextRow());
+    }
+
+    if (QueryResult result = WorldDatabase.Query("SELECT ContentType, ItemType, Entry, Price, Enchant, EnchantSlot, Achievement, Enabled FROM zone_difficulty_hardmode_rewards"))
+    {
+        /* debug
+         * uint32 i = 0;
+         * end debug
+         */
+        do
+        {
+            /* debug
+             * ++i;
+             * end debug
+             */
+            ZoneDifficultyRewardData data;
+            uint32 contenttype = (*result)[0].Get<uint32>();
+            uint32 itemtype = (*result)[1].Get<uint32>();
+            data.Entry = (*result)[2].Get<uint32>();
+            data.Price = (*result)[3].Get<uint32>();
+            data.Enchant = (*result)[4].Get<uint32>();
+            data.EnchantSlot = (*result)[5].Get<uint8>();
+            data.Achievement = (*result)[6].Get<uint32>();
+            bool enabled = (*result)[7].Get<bool>();
+
+            if (enabled)
+            {
+                sZoneDifficulty->Rewards[contenttype][itemtype].push_back(data);
+                //LOG_INFO("sql", "Loading item with entry {} has enchant {} in slot {}. contenttype: {} itemtype: {}", data.Entry, data.Enchant, data.EnchantSlot, contenttype, itemtype);
+            }
+            //LOG_INFO("sql", "Total items in vector: {}.", i);
+        } while (result->NextRow());
+    }
+}
+
+/* Loads the HardmodeInstanceData from the database.
+*  Fetch from zone_difficulty_instance_saves.
+*
+*  `InstanceID` INT NOT NULL DEFAULT 0,
+*  `HardmodeOn` TINYINT NOT NULL DEFAULT 0,
+*  `HardmodePossible` TINYINT NOT NULL DEFAULT 1,
+*
+*  Exclude data not in the IDs stored in GetInstanceIDs() and delete
+*  zone_difficulty_instance_saves for instances that no longer exist.
+*/
+void ZoneDifficulty::LoadHardmodeInstanceData()
+{
+     std::vector<bool> instanceIDs = sMapMgr->GetInstanceIDs();
+     /* debugging
+     * for (int i = 0; i < int(instanceIDs.size()); i++)
+     * {
+     *   LOG_INFO("sql", "ZoneDifficulty::LoadHardmodeInstanceData: id {} exists: {}:", i, instanceIDs[i]);
+     * }
+     * end debugging
+     */
+    if (QueryResult result = CharacterDatabase.Query("SELECT * FROM zone_difficulty_instance_saves"))
+    {
+        do
+        {
+            uint32 InstanceId = (*result)[0].Get<uint32>();
+            bool HardmodeOn = (*result)[1].Get<bool>();
+            bool HardmodePossible = (*result)[2].Get<bool>();
+
+            if (instanceIDs[InstanceId] == true)
```

Comment:
no need for == true. It's a bool statement, can only be true or false

if (instanceIDs[InstanceId]) is the same as if (instanceIDs[InstanceId] == true)

Reply:
you never need to specify true/false, only when you are looking for stuff that isnt just a bool